tests/unit/io/test_twistedreactor.py

```diff
@@ -66,7 +66,9 @@ def setUp(self):
         self.tr.protocol = self.obj_ut
 
     def tearDown(self):
-        pass
+        loop = twistedreactor.TwistedConnection._loop
+        if not loop._reactor_stopped():
+            loop._cleanup()
```

Comment:
This test started failing after moving to pytest, apparently because the order of operations changed.  It looks like the old infrastructure included implicit assumptions about when the reactor was started... and now that pytest executed tests in a different order those assumptions led to unexpected test failures.  The goal of this change is to make the test more robust by forcing each test case to do a better job cleaning up after itself.